e2e/tests/core/02-client/client_test.go

```diff
@@ -65,8 +66,16 @@ func (s *ClientTestSuite) TestScheduleIBCUpgrade_Succeeds() {
 	t := s.T()
 	ctx := context.TODO()
 
-	_, _ = s.SetupChainsRelayerAndChannel(ctx, nil)
 	chainA, chainB := s.GetChains()
+	chainAVersion := chainA.Config().Images[0].Version
+
+	transferVersion := transfertypes.V1
+	if testvalues.ICS20v2FeatureReleases.IsSupported(chainAVersion) {
+		transferVersion = transfertypes.V2
+	}
```

Comment:
could we maybe add this logic [here](https://github.com/cosmos/ibc-go/blob/bc446f9c484ca7374f4c34ecf3a6c8dd8a87b870/e2e/testsuite/testsuite.go#L159) ?

From there we have access to the chains and can determine the version we should use as the default. We can then override it as needed if the test requires.

Reply:
Yes, good point. I will adjust it, it might be easier to start from a new PR 😅